crates/sdk/src/db_connection.rs

```diff
@@ -877,6 +881,16 @@ but you must call one of them, or else the connection will never progress.
         self
     }
 
+    /// Sets the compression used when a certain threshold in the message size has been reached.
+    ///
+    /// The current threshold used by the host is 1KiB for the entire server message
+    /// and for individual query updates.
+    /// Note however that this threshold is not guaranteed and may change without notice.
+    pub fn with_compression(mut self, compression: Compression) -> Self {
+        self.compression = compression;
+        self
+    }
```

Comment:
I don't love exposing this here, honestly. I'm not sure we're prepared to commit to the set of compression algorithms we expose, the default, or to what extent the host promises to comply with the client's requested format. I also don't honestly see a strong need for it - as I understand it we wanted to offer GZip because JS engines have it built-in, so we remove a dependency from the TypeScript client SDK and likely get more optimized decompression there. Neither of these things apply to the Rust client SDK, so I don't see the value in exposing this. Notably I think we actively do not want people choosing `Compression::None`.

All this to say, I would feel better leaving this method unstable for now.

```suggestion
    #[doc(hidden)]
    /// Sets the compression used when a certain threshold in the message size has been reached.
    ///
    /// The current threshold used by the host is 1KiB for the entire server message
    /// and for individual query updates.
    /// Note however that this threshold is not guaranteed and may change without notice.
    ///
    /// This method is unstable and is subject to change incompatibly without a major version bump.
    pub fn with_compression(mut self, compression: Compression) -> Self {
        self.compression = compression;
        self
    }
```